addons/l10n_my_edi_extended/tests/test_file_generation.py

```diff
@@ -0,0 +1,96 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from freezegun import freeze_time
+from lxml import etree
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.addons.l10n_my_edi.tests.test_file_generation import NS_MAP
+from odoo.tests import tagged
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class L10nMyEDITestFileGeneration(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref='my'):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        # TIN number is required
+        cls.company_data['company'].write({
+            'vat': 'C2584563200',
+            'l10n_my_edi_mode': 'test',
+            'l10n_my_edi_industrial_classification': cls.env['l10n_my_edi.industry_classification'].search([('code', '=', '01111')]).id,
+            'l10n_my_identification_type': 'BRN',
+            'l10n_my_identification_number': '202001234567',
+            'state_id': cls.env.ref('base.state_my_jhr').id,
+            'street': 'that one street, 5',
+            'city': 'Main city',
+            'phone': '+60123456789',
+        })
+        cls.partner_b.write({
+            'vat': '123456789',
+            'l10n_my_identification_type': 'BRN',
+            'l10n_my_identification_number': 'NA',
+            'country_id': cls.env.ref('base.us').id,
+            'state_id': cls.env.ref('base.state_us_1'),
+            'street': 'that other street, 3',
+            'city': 'Main city',
+            'phone': '+60123456785',
+            'l10n_my_edi_malaysian_tin': 'EI00000000020',
+        })
+        cls.product_a.l10n_my_edi_classification_code = "001"
+
+        cls.purchase_tax = cls.env['account.tax'].create({
+            'name': 'tax_10',
+            'amount_type': 'percent',
+            'amount': 10,
+            'type_tax_use': 'purchase',
+            'country_id': cls.env.ref('base.my').id,
+        })
+
+    @freeze_time('2024-07-15 10:00:00')
+    def test_07_self_billing(self):
+        bill = self.init_invoice(
+            'in_invoice', partner=self.partner_b, products=self.product_a, taxes=self.purchase_tax,
+        )
+        bill.action_post()
+
+        file, errors = bill._l10n_my_edi_generate_invoice_xml()
+        self.assertEqual(errors, set())
```

Comment:
Do you care about it being a set?
```suggestion
        self.assertFalse(errors)
```